src/EscapeAnalysis.jl

```diff
@@ -370,18 +371,18 @@ end
 function escape_builtin!(::typeof(tuple), args::Vector{Any}, pc::Int, state::EscapeState, ir::IRCode, changes::Changes)
     info = state.ssavalues[pc]
     info === NoInformation() && (info = NoEscape())
-    for arg in args[2:end]
-        push!(changes, (arg, info))
-    end
+    update_non_bitstype_changes!(args, ir, info, changes)
     return true
 end
 
 # TODO don't propagate escape information to the 1st argument, but propagate information to aliased field
 function escape_builtin!(::typeof(getfield), args::Vector{Any}, pc::Int, state::EscapeState, ir::IRCode, changes::Changes)
     info = state.ssavalues[pc]
     info === NoInformation() && (info = NoEscape())
-    for arg in args[2:end]
-        push!(changes, (arg, info))
+    field_type = widenconst(ir.stmts.type[pc])
```

Comment:
nitpick: I prefer
```suggestion
   rt = widenconst(ir.stmts.type[pc])
```
because of the existence of `getfield(::Module, ::Symbol)`.

Not too stick with this, though. Your choice :)